### ISSUE TITLE
QGIS 2.18.10 will not launch or install

### ISSUE THREAD

**Issue by ReneMalenfant** [OSGeo/homebrew-osgeo4mac#255]
A number of the osgeo4mac bottles were updated earlier today, which has rendered QGIS 2.18.10 unusable (and I could not reinstall it).  The gist is here:
https://gist.github.com/ReneMalenfant/87b3edd8758584d02e4fa93432b93bb1

It appears to be related to QScintilla.  (/usr/local/opt/qscintilla2-qt4/ does not have an 'include' directory. I fixed the problem by symlinking it to /usr/local/opt/qscintilla2-qt4/libexec/include, although it died again with a compilation error, I think because /usr/local/opt/qscintilla2-qt4/lib does not contain a symlink to libqscintilla2.12.0.2.dylib)

**Comment by dakcarto** [comment]
Hi @ReneMalenfant, thanks for reporting!

The `qgis*` formulae have not yet been updated to work with those dependencies (or sandboxing). Working on that today. They will be broken until fixes are applied.

**Comment by ReneMalenfant** [comment]
Okay. Thanks, dakcarto.

**Comment by luisspuerto** [comment]
I also have this problem...

**Comment by luisspuerto** [comment]
I also would like to know why this things happens. Shouldn't we update that often?

**Comment by dakcarto** [comment]
@luisspuerto, Homebrew is similar to an 'unstable' Linux repo, where things can go through spells of broken-ness (see #257 for how fast things can become broken due to overzealous version bumping at Homebrew-core).

I should have the `qgis*` formulae finished today. I have them building locally; just trying to get Travis CI to not timeout on QGIS builds.

This is an open source Homebrew tap and you are welcome to submit PRs to help out.

**Comment by eldang** [comment]
Thank you @dakcarto for your work on this.

Is there a way that those of us impacted by it can roll back to a last known good version?  I'd like to do something more constructive than that, but frankly I'm using homebrew because I don't have the knowledge to debug complex install processes myself.

**Comment by eldang** [comment]
Sorry, I was too quick to give up and have now figured out what to roll back and how.  Sharing this in the hope that it helps others who have this problem having had a working QGIS before the update:

The faulty version of QScintilla appears to be qscintilla2-qt4/2.9.3_2 .  
So first check if you have any previous version installed with `brew info qscintilla2-qt4`
On my machine, the output from that showed me both the faulty version (built today), and 2.9.3_1 (built last October).
If you have a previous version available, you can switch to it with `brew switch qscintilla2-qt4 2.9.3_1` (or whichever version number you have)
I then got the following warning in the output from `brew switch`:
```Error: Could not symlink include/Qsci/qsciabstractapis.h
Target /usr/local/include/Qsci/qsciabstractapis.h
is a symlink belonging to qscintilla2. You can unlink it:
  brew unlink qscintilla2

To force the link and overwrite all conflicting files:
  brew link --overwrite qscintilla2-qt4
```

I used the `brew link --overwrite qscintilla2-qt4` solution and with that my previously installed version of QGIS is back in working order.

If you don't have a previous version of QScintilla present, or are installing QGIS from scratch, then you'll need to download the older formulae, which is a step beyond what I know how to do right now.

**Comment by joaoponceleao** [comment]
I also can't get it to build on sierra.
It stops here:
```
[ 92%] Built target qgis_app
make: *** [all] Error 2
```
The only thing I managed to find in the console logs is this file not found error:
```
In file included from /tmp/qgis2-20170814-16216-fkn3z/QGIS-final-2_18_11/src/customwidgets/../gui/qgscodeeditorpython.h:19:
/tmp/qgis2-20170814-16216-fkn3z/QGIS-final-2_18_11/src/customwidgets/../gui/qgscodeeditor.h:22:10: fatal error: 'Qsci/qsciapis.h' file not found
#include <Qsci/qsciapis.h>
``` 
and:
```
1 error generated.
make[2]: *** [src/customwidgets/CMakeFiles/qgis_customwidgets.dir/qgsexpressionbuilderwidgetplugin.cpp.o] Error 1
make[1]: *** [src/customwidgets/CMakeFiles/qgis_customwidgets.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 82%]
```
But that happens before it reaches here:
```
[ 92%] Built target qgis_app
make: *** [all] Error 2
```
